omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/diagnosis/EncounterDiagnosesFragmentController.java

```diff
@@ -54,7 +54,7 @@ public int compare(Diagnosis left, Diagnosis right) {
                 SimpleObject simple = diagnosesFragmentController.simplify(csr, ui, Context.getLocale());
                 jsDiagnosis = simple.toJson();
             }
-            jsForDiagnoses.add("{ diagnosis: diagnoses.CodedOrFreeTextConceptAnswer(" + jsDiagnosis + "), confirmed: " + (d.getCertainty().equals(Diagnosis.Certainty.CONFIRMED)) + ", primary: " + (d.getOrder().equals(Diagnosis.Order.PRIMARY)) + ", existingObs: " + (d.getExistingObs() != null ? d.getExistingObs().getId() : null) + " }");
+            jsForDiagnoses.add("{ diagnosis: diagnoses.CodedOrFreeTextConceptAnswer(" + jsDiagnosis + "), confirmed: " + (d.getCertainty().equals(Diagnosis.Certainty.CONFIRMED)) + ", primary: " + (d.getOrder().equals(Diagnosis.Order.PRIMARY)) + ", existingObs: " + (d.getExistingObs() != null ? d.getExistingObs().getId() : null) + ", existingDiagnosis: " + d.getExistingDiagnosis() + " }");
```

Comment:
Did you test this for cases where existing diagnosis returns null?

Reply:
@dkayiwa is [this](https://github.com/pluwum/openmrs-module-coreapps/blob/aa34c02b443e4b63acf55966e15d1511a727edc0/omod/src/test/java/org/openmrs/module/coreapps/fragment/controller/diagnosis/EncounterDiagnosesFragmentControllerTest.java#L60) what you mean?

Reply:
Looks ok.